lib/node_modules/@stdlib/string/base/invcase/lib/main.js

```diff
@@ -0,0 +1,60 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2022 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var uppercase = require( '@stdlib/string/base/uppercase' );
+var lowercase = require( '@stdlib/string/base/lowercase' );
+var trim = require( '@stdlib/string/base/trim' );
+
+// MAIN //
+
+/**
+* Converts a string to constant case.
+*
+* @param {string} str - string to convert
+* @returns {string} constant-cased string
+*
+* @example
+* var str = invcase( 'beep' );
+* // returns 'BEEP'
+*
+* @example
+* var str = invcase( 'beep BOOP' );
+* // returns 'BEEP boop'
+*
+* @example
+* var str = invcase( 'isMobile' );
+* // returns 'IS mOBILE'
+*
+* @example
+* var str = invcase( 'HeLlO wOrLd!' );
+* // returns 'hElLo WoRlD!'
+*/
+function invcase( str ) {
+	str = trim( str );
+    str = [...str].map(ch => ch === uppercase( ch ) ? lowercase( ch ) : uppercase( ch )).join('');
+	return str;
```

Comment:
```suggestion
	var out;
	var ch;
	var s;
	var i;
	
	out = [];
	for ( i = 0; i < str.length; i++ ) {
		ch = str[ i ];
		s = uppercase( ch );
		if ( s === ch ) {
			s = lowercase( ch );
		}
		out.push( s );
	}
	return out.join( '' );
```

We typically prefer imperative code as (1) we avoid call overhead and (2) it is generally faster.

We don't need to `trim` here, as, in this case, I think that might be unintuitive. For other things like camel, Pascal, and snake case, removing whitespace makes sense. Here, I'd assume that whitespace would remain.